src/Panel/Controller/Search.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Kirby\Panel\Controller;
+
+use Kirby\Cms\App;
+use Kirby\Toolkit\Escape;
+
+/**
+ * The Search controller takes care of the logic
+ * for delivering Panel search results
+ *
+ * @package   Kirby Panel
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ * @internal
+ */
+class Search
+{
+	public static function files(
+		string $query,
+		int|null $limit = null,
+		int $page = 1
+	): array {
+		$kirby = App::instance();
+		$files = $kirby->site()
+			->index(true)
+			->filter('isListable', true)
```

Comment:
```suggestion
```
I think this filter is not needed since we'll filter again after few lines later.

Reply:
This would be a change to how the code was before. I think it's really meant to only use files from listable pages and then filter for listable files. One filter is applied to pages, the other to the files.

Reply:
Oh that's right. I missed the pages and files collection there.